src/main/java/org/codehaus/mojo/versions/UseReleasesMojo.java

```diff
@@ -103,13 +124,269 @@ protected void update( ModifiedPomXMLEventReader pom )
+                + " is not available for the following dependencies: " + noReleaseForArtifacts );
+            if ( failIfNotReplaced )
+            {
+                throw new NoSuchElementException( "No matching release of " + currentVersion + " found for dependencies " + noReleaseForArtifacts );
+            }
+        }
+    }
+
+    /**
+     * Replace the property with the release version which match the given range and which is common to all dependencies refering to the property. If no common release version is available, do nothing.
+     *
+     * @param pom
+     * @param propertyVersions
+     * @param currentVersion
+     * @param releaseVersion
+     * @throws MojoExecutionException
+     * @throws ArtifactMetadataRetrievalException
+     * @throws XMLStreamException
+     */
+    private void propertyRangeMatching( ModifiedPomXMLEventReader pom, PropertyVersions propertyVersions, String currentVersion, String releaseVersion )
+        throws ArtifactMetadataRetrievalException, MojoExecutionException, XMLStreamException
+    {
+        ArtifactVersion[] commonVersions = null;
+        for ( ArtifactAssociation artifactAssociation : propertyVersions.getAssociations() )
+        {
+            if ( artifactAssociation.getArtifact().getVersion() != null )
+            {
+                artifactAssociation.getArtifact().setVersion( releaseVersion );
+                ArtifactVersions artifactVersions = getHelper().lookupArtifactVersions( artifactAssociation.getArtifact(), false );
+                if ( !isExcludeReactor() )
+                {
+                    artifactVersions = mergeReactorArtifactsVersions( artifactVersions );
+                }
+                if ( commonVersions == null )
+                {
+                    commonVersions = artifactVersions.getVersions();
+                }
+                else
+                {
+                    commonVersions = intersection( commonVersions, artifactVersions.getVersions() );
+                }
+                Arrays.sort( commonVersions, Collections.reverseOrder( artifactVersions.getVersionComparator() ) );
+            }
+        }
+
+        if ( commonVersions != null && commonVersions.length != 0 )
+        {
+            ArtifactVersion finalVersion = null;
+            for ( ArtifactVersion proposedVersion : commonVersions )
+            {
+                if ( proposedVersion.toString().startsWith( releaseVersion ) )
+                {
+                    getLog().debug( "Found matching release version " + proposedVersion + " for property " + propertyVersions.getName() + " with range version " + releaseVersion );
+                    finalVersion = proposedVersion;
+                    break;
+                }
+            }
+
+            if ( PomHelper.setPropertyVersion( pom, propertyVersions.getProfileId(), propertyVersions.getName(), finalVersion.toString() ) )
```

Comment:
Hi,

"finalVersion" may be uninitialized here.
I tried to use your changes and caught NullPointer exception with my pom's:

Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.codehaus.mojo:versions-maven-plugin:2.8-SNAPSHOT:use-releases failed.
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
	at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:207)
	... 21 more
Caused by: java.lang.NullPointerException
	at org.codehaus.mojo.versions.UseReleasesMojo.propertyRangeMatching(UseReleasesMojo.java:286)
	at org.codehaus.mojo.versions.UseReleasesMojo.useReleaseProperties(UseReleasesMojo.java:179)
	at org.codehaus.mojo.versions.UseReleasesMojo.update(UseReleasesMojo.java:129)
	at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.process(AbstractVersionsUpdaterMojo.java:314)
	at org.codehaus.mojo.versions.AbstractVersionsUpdaterMojo.execute(AbstractVersionsUpdaterMojo.java:250)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 22 more

Regards,
Sergei

Reply:
Fixed. Thank you for the feedback.